### PR TITLE
[DeviceSanitizer] Only allows to report one error per each kernel launch

### PR DESCRIPTION
**Pull request by zhaomaosu**
When error is detected, the corresponding work-item will exit without unpoisoning local/private shadow memory. Then other work-items are easy to cause false alarms. So, we only allows to report one error per each kernel launch by default.

UR Part: https://github.com/oneapi-src/unified-runtime/pull/2069

### CODE REVIEW

**Comment by AllanZyne** [comment]
How about skipping sanitizer when "UR_SANITIZER_INCLUDE_DIR" is empty string?

**Comment by zhaomaosu** [comment]
> How about skipping sanitizer when "UR_SANITIZER_INCLUDE_DIR" is empty string?

Done.

**Review by yingcong-wu — APPROVED** [review]

**Comment by pbalcer** [comment]
please rebase (and ping me, I want to merge this asap)

**Comment by zhaomaosu** [comment]
> please rebase (and ping me, I want to merge this asap)

@pbalcer, thanks, done.

**Review by yingcong-wu — APPROVED** [review]

**Comment by pbalcer** [comment]
@intel/llvm-reviewers-runtime can you please review? thanks.

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 1]
```diff
@@ -116,7 +116,7 @@ if(SYCL_UR_USE_FETCH_CONTENT)
       CACHE PATH "Path to external '${name}' adapter source dir" FORCE)
   endfunction()
 
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
+  set(UNIFIED_RUNTIME_REPO "https://github.com/zhaomaosu/unified-runtime.git")
```
Comment (pbalcer) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:119]: to be updated after UR PR merges.

**Review by pbalcer — APPROVED** [review]

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake (hunk 1 above)
Comment (steffenlarsen) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:119]: When there are testing redirects like this, I would prefer that unified runtime reviewers don't give a formal approval until it has been addressed. We risk that the gatekeepers miss a comment like this and merge.

**Review by steffenlarsen — APPROVED** [review]
LGTM! We need the repo redirect addressed before we merge however.

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake (hunk 1 above)
Comment (pbalcer) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:119]: Ok, will keep that in mind.

**Comment by pbalcer** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**